lib/uiautomator.js

```diff
@@ -63,8 +63,10 @@ class UiAutomator extends events.EventEmitter {
 
   async shutdown () {
     log.debug('Shutting down UiAutomator');
-    this.changeState(UiAutomator.STATE_STOPPING);
-    await this.proc.stop();
+    if (this.state !== UiAutomator.STATE_STOPPED) {
+      this.changeState(UiAutomator.STATE_STOPPING);
+      await this.proc.stop();
+    }
     await this.killUiAutomatorOnDevice();
```

Comment:
should this block also be executed if the current state is already STATE_STOPPED?

Reply:
also, should it method call be awaited?

Reply:
This is what I was referring to here: "I wasn't sure what to do about the killing of leaked uiautomator processes. It doesn't happen when it detects an unexpected termination of the process, so I left it in, in the showtdown."